src/mem/cache/prefetch/stride.cc

```diff
@@ -169,17 +174,15 @@ Stride::calculatePrefetch(const PrefetchInfo &pfi,
 
         entry->lastAddr = pf_addr;
 
+        // Return if we don't see a stride match
```

Comment:
I think this would be a bug. The point of having the confidence threshold is exactly to decide when the predicted stride is relevant or not. If we have 99% confidence that an access to X will have a stride S, but this time we got a stride of S2, then the confidence indicates that S2 was just an outlier, and that we will probably see S again in the next access.

Reply:
I agree. I've changed the code to reflect this